dev/com.ibm.ws.jbatch.security/src/com/ibm/ws/jbatch/security/impl/WSBatchAuthServiceImpl.java

```diff
@@ -150,25 +175,84 @@ public long authorizedInstanceRead(long instanceId)
         throws JobSecurityException, NoSuchJobInstanceException {
 
         String submitter = persistenceManagerService.getJobInstanceSubmitter(instanceId);
+        Set<String> listOfGroupsForJobID = null;
+        Set<String> listOfGroupsForSubject = null;
+        boolean checkGroupSecurity = false;
 
         if (this.isAdmin(runAsSubject())) {
         } else if (this.isMonitor(runAsSubject())) {
         } else if (this.isSubmitter(runAsSubject())) {
+        	boolean submitterRole = this.isSubmitter(runAsSubject());
+        	boolean isGroupAdmin = this.isGroupAdmin();
+        	boolean isGroupMonitor = this.isGroupMonitor();
             //if you don't also own the job you can't view it
-            if (!batchSecurityHelper.getRunAsUser().equals(submitter)){
-                throw new JobSecurityException( getFormattedMessage( "USER_UNAUTHORIZED_JOB_INSTANCE",
-                                                                     new Object[] {getRunAsUser(), instanceId},
-                                                                     "CWWKY0302W: User {0} is not authorized to perform batch operations associated with job instance {1}.") );
-            }
-        } else {
+			if (batchSecurityHelper.getRunAsUser().equals(submitter)) {
+			} else if (this.isGroupAdmin() || this.isGroupMonitor()) {
+				checkGroupSecurity = true;
+			} else {
+
+				throw new JobSecurityException(getFormattedMessage("USER_UNAUTHORIZED_JOB_INSTANCE",
+						new Object[] { getRunAsUser(), instanceId },
+						"CWWKY0302W: User {0} is not authorized to perform batch operations associated with job instance {1}."));
+			}
+        } else if (this.isGroupAdmin() || this.isGroupMonitor()) { 
+    		checkGroupSecurity = true;
+    	}
+        else {
             throw new JobSecurityException( getFormattedMessage( "USER_UNAUTHORIZED_NO_BATCH_ROLES",
                                                                  new Object[] { getRunAsUser() },
                                                                  "CWWKY0303W: User {0} is not authorized to perform any batch operations." ) );
         }
-
+        	      	
+		if (checkGroupSecurity) {
+			
+			listOfGroupsForJobID = persistenceManagerService.getJobInstance(instanceId).getGroupNames();
+			listOfGroupsForSubject = getSubjectGroups(runAsSubject());
+			
+			if (subjectInGroups(listOfGroupsForSubject, listOfGroupsForJobID)) {
+			} else {
+				// user not in any groups listed - disallow access
+				// construct message to be displayed
+
+				String log_jobGroups = constructGroupListForAuthFailString(listOfGroupsForJobID);
```

Comment:
You might not need the `constructGroupListForAuthFailString` method .. the Set **toString()** might be enough?